internal/buffer/search.go

```diff
@@ -6,38 +6,45 @@ import (
 	"github.com/zyedidia/micro/v2/internal/util"
 )
 
-func padRegexp(r *regexp.Regexp) (*regexp.Regexp, *regexp.Regexp, *regexp.Regexp, *regexp.Regexp) {
-	rPadStart := regexp.MustCompile(".(?:"+r.String()+")")
-	rPadEnd := regexp.MustCompile("(?:"+r.String()+").")
-	rPadBoth := regexp.MustCompile(".(?:"+r.String()+").")
-	return r, rPadStart, rPadEnd, rPadBoth
+const (
+	rPadNone = iota
+	rPadStart
+	rPadEnd
+	rPadBoth
+)
+
+func padRegexp(r *regexp.Regexp) [4]*regexp.Regexp {           // r is used when searching an entire line
```

Comment:
I didn't mean that we should explain when is each of these regexes used (that is rather clear from the code). I meant that we should explain _why_ are we even doing all this. It is not obvious that we are doing all this just to avoid false matches for `^`, `$`, `\b` etc.

Basically you already explained that in the PR description, so my point is that we should also explain it in the code. Otherwise readers of the code are gonna yell "WTF is this?" and remain confused and frustrated until they run `git blame` and then go to the github webpage of this PR and read the PR description (since the commit message doesn't include this explanation either).

Reply:
Check out the new comment. It's another commit; we can squash when it's good to go.